docker/latest/Dockerfile

```diff
@@ -30,56 +32,79 @@ RUN chmod +x /bin/bash_run \
+      && echo "deb [arch=amd64 signed-by=/usr/share/keyrings/cuda-archive-keyring.asc] https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/x86_64 /" > /etc/apt/sources.list.d/cuda.list \
+      && wget -O /etc/apt/preferences.d/cuda-repository-pin-600 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/x86_64/cuda-ubuntu2404.pin \
+ ;; esac \
  && echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/pdidev-archive-keyring.gpg] https://raw.githubusercontent.com/pdidev/repo/ubuntu noble main" > /etc/apt/sources.list.d/pdi.list \
  && wget -q -O /etc/apt/keyrings/pdidev-archive-keyring.gpg https://raw.githubusercontent.com/pdidev/repo/ubuntu/pdidev-archive-keyring.gpg \
  && apt-get update -y \
  && apt-get install -y --no-install-recommends \
     build-essential \
-    cmake \
     git \
     libfftw3-dev \
     liblapacke-dev \
     libhwloc-dev \
     libpdi-dev \
     pdidev-archive-keyring \
     pkg-config \
+ # Installing cmake < 3.28 to workaround issue with Kokkos
+ && wget https://github.com/Kitware/CMake/releases/download/v3.27.9/cmake-3.27.9-linux-x86_64.tar.gz \
+ && tar -xvf cmake-3.27.9-linux-x86_64.tar.gz --directory /opt \
+ && rm -f cmake-3.27.9-linux-x86_64.tar.gz \
  && case "${BACKEND}" in \
     "cpu") \
       apt-get install -y --no-install-recommends \
       clang \
       clang-tidy \
       clang-format \
  ;; "cuda") \
+      # libnvjitlink-12-6 seems to be a missing dependency of cusparse
       apt-get install -y --no-install-recommends \
-      nvidia-cuda-toolkit \
+      cuda-minimal-build-12-6 \
+      cuda-nvtx-12-6 \
+      libcublas-dev-12-6 \
+      libcufft-dev-12-6 \
+      libcurand-dev-12-6 \
+      libcusparse-dev-12-6 \
+      libnvjitlink-12-6 \
       gcc-12 \
       g++-12 \
  ;; "hip") \
       apt-get install -y --no-install-recommends \
-      rocm-hip-sdk \
+      # libxml2 seems to be a missing dependency of rocm-hip-runtime
+      libxml2 \
+      hipblas-dev${ROCM_VERSION}.0 \
+      hipfft-dev${ROCM_VERSION}.0 \
+      hiprand-dev${ROCM_VERSION}.0 \
+      hipsparse-dev${ROCM_VERSION}.0 \
+      rocm-hip-runtime-dev${ROCM_VERSION}.0 \
+      rocrand-dev${ROCM_VERSION}.0 \
+      rocthrust-dev${ROCM_VERSION}.0 \
       && echo "/opt/rocm/lib" >> /etc/ld.so.conf.d/rocm.conf \
       && echo "/opt/rocm/lib64" >> /etc/ld.so.conf.d/rocm.conf \
       && ldconfig \
-;; esac \
+ ;; esac \
  && git clone -b v1.8.0 https://github.com/ginkgo-project/ginkgo.git \
  && cd ginkgo \
  && case "${BACKEND}" in \
     "cpu") \
-      cmake -S . -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DGINKGO_BUILD_OMP=ON -DGINKGO_BUILD_TESTS=OFF -DGINKGO_BUILD_EXAMPLES=OFF -DGINKGO_BUILD_BENCHMARKS=OFF \
+      cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DGINKGO_BUILD_OMP=ON -DGINKGO_BUILD_TESTS=OFF -DGINKGO_BUILD_EXAMPLES=OFF -DGINKGO_BUILD_BENCHMARKS=OFF \
```

Comment:
I would like to install ginkgo during GitHub actions not inside a Dockerfile. If there is a reason, could you please add comments?

Reply:
It takes a long time to compile, more than 30 min. So we compile it once and for all in the images and use it the CI.

Reply:
Once we install ginkgo during actions, we can use it afterwards as well as other libraries. I guess other libraries are also compiled once and used in CI. Why you need special treatments for ginkgo?

Reply:
I don't think there is a special treatment for Ginkgo. Most of our dependencies are installed with distribution packages in the image. Actually I think the special treatment is for Kokkos/Kokkos-Kernels/benchmark/googletest. In my opinion these latter should belong to the image. Did I understand correctly ?

Reply:
Actually I just thought the treatment for `Kokkos/Kokkos-Kernels/benchmark/googletest` is usual. If you would like to install these libraries in the image as well as `ginkgo`, that is fine. My question is why you install `ginkgo` in a different way than others.

Reply:
I think it is kind of legacy. Before using Ginkgo, we were building Kokkos/benchmark/googletest when building DDC. It was fast enough. However when we introduced Ginkgo it was not possible to use this strategy as Ginkgo takes much more time than others. We then moved Ginkgo directly in the image but not the others. It should be ok to also move the remaining dependencies in the image.

Reply:
Sure, then it is OK to keep as is.
I prefer not to install too much inside an image, but if the installation takes too long, we have no choice.